app/src/main/java/de/danoeh/antennapod/dialog/SubscriptionFilterDialog.java

```diff
@@ -0,0 +1,88 @@
+package de.danoeh.antennapod.dialog;
+
+import android.content.Context;
+import android.view.View;
+import android.widget.ArrayAdapter;
+import android.widget.Spinner;
+import androidx.appcompat.app.AlertDialog;
+
+import java.util.ArrayList;
+
+import de.danoeh.antennapod.R;
+import de.danoeh.antennapod.core.feed.SubscriptionFilter;
+
+import static de.danoeh.antennapod.core.feed.SubscriptionFilter.STRING_ALWAYS;
+import static de.danoeh.antennapod.core.feed.SubscriptionFilter.STRING_DISABLE;
+import static de.danoeh.antennapod.core.feed.SubscriptionFilter.STRING_ENABLE;
+import static de.danoeh.antennapod.core.feed.SubscriptionFilter.STRING_GLOBAL_DEFAULT;
+import static de.danoeh.antennapod.core.feed.SubscriptionFilter.STRING_NEVER;
+import static de.danoeh.antennapod.core.feed.SubscriptionFilter.STRING_NO_FILTER;
+
+public abstract class SubscriptionFilterDialog {
+
+    private Context context;
+    private Spinner spinnerAutoDownload;
+    private Spinner spinnerKeepUpdated;
+    private Spinner spinnerAutoDelete;
+    private SubscriptionFilter.AutoDownload autoDownload;
+    private SubscriptionFilter.KeepUpdated keepUpdated;
+    private SubscriptionFilter.AutoDelete autoDelete;
+    private AlertDialog dialog;
+    protected SubscriptionFilterDialog(Context theContext, SubscriptionFilter.AutoDownload theAutoDownload, SubscriptionFilter.KeepUpdated theKeepUpdated,
+                                       SubscriptionFilter.AutoDelete theAutoDelete) {
+        this.context = theContext;
+        this.autoDownload = theAutoDownload;
+        this.keepUpdated = theKeepUpdated;
+        this.autoDelete = theAutoDelete;
+
+    }
+
+    public AlertDialog openDialog() {
+
+        View content = View.inflate(context, R.layout.dialog_layout, null);
+        AlertDialog.Builder builder = new AlertDialog.Builder(context);
+
+        builder.setView(content);
+        spinnerAutoDownload = content.findViewById(R.id.spinnerAutoDownload);
+        spinnerKeepUpdated = content.findViewById(R.id.spinnerKeepUpdated);
+        spinnerAutoDelete = content.findViewById(R.id.spinnerAutoDelete);
+
+        ArrayList<String> spinnerArrayAutoDown_keepUpdated = new ArrayList<>();
+        ArrayList<String> spinnerArrayAutoDelete = new ArrayList<>();
+
+        spinnerArrayAutoDown_keepUpdated.add(STRING_ENABLE);
+        spinnerArrayAutoDown_keepUpdated.add(STRING_DISABLE);
+        spinnerArrayAutoDown_keepUpdated.add(STRING_NO_FILTER);
```

Comment:
Strings that are shown to the user should come from `R.strings.xy`. In this case, it might be a good idea to use a string array that references strings. Those strings then could be added to `strings.xml`.

Something like this:
https://github.com/AntennaPod/AntennaPod/blob/develop/core/src/main/res/values/arrays.xml#L4-L8